pkg/build/controller/strategy/sti.go

```diff
@@ -31,7 +31,7 @@ var STITempDirectoryCreator = &tempDirectoryCreator{}
 // CreateBuildPod creates a pod that will execute the STI build
 // TODO: Make the Pod definition configurable
 func (bs *STIBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	buildJSON, err := json.Marshal(build)
+	data, err := v1beta1.Codec.Encode(build)
```

Comment:
Inject the codec from the BuildControllerFactory through the typeBasedFactoryStrategy.